packages/gatsby/src/query/index.js

```diff
@@ -92,18 +92,16 @@ const processPageQueries = async (
   // /dev-404-page/, whose SitePage node is created via
   // `internal-data-bridge`, but the actual page object is only
   // created during `gatsby develop`.
-  const pages = _.filter(queryIds.map(id => state.pages.get(id)))
-  await processQueries(
-    pages.map(page => createPageQueryJob(state, page)),
-    {
-      activity,
-      graphqlRunner,
-      graphqlTracing,
-    }
-  )
+  const jobs = queryIds.map(id => createPageQueryJob(state, id))
+  await processQueries(jobs, {
+    activity,
+    graphqlRunner,
+    graphqlTracing,
+  })
 }
 
-const createPageQueryJob = (state, page) => {
+const createPageQueryJob = (state, pageId) => {
+  const page = state.pages.get(pageId)
```

Comment:
If the page was deleted after it was added to a queue - then the `page` can be undefined here. I guess the point of filter was to remove all `undefined` values

Reply:
You know what. You're right. I was thinking the identity predicate would work like map. dang!

Reply:
I think we can still avoid the unnecessary filter loop - we just need to replace `queryIds.map` with some `for` loop and continue when the page is undefined.